Sources/PortalSwift/Core/Api/PortalApi.swift

```diff
@@ -0,0 +1,589 @@
+//
+//  PortalApi.swift
+//
+//  Created by Portal Labs, Inc.
+//  Copyright © 2022 Portal Labs, Inc. All rights reserved.
+//
+
+import Foundation
+
+/// The class to interface with Portal's REST API.
+public class PortalApi {
+  private var _client: ClientResponse?
+  private var apiKey: String
+  private var baseUrl: String
+  private let decoder = JSONDecoder()
+  private var httpRequests: HttpRequester
+  private let logger = PortalLogger()
+  private var provider: PortalProvider?
+  private let requests: PortalRequests
+  private let featureFlags: FeatureFlags?
+
+  private var address: String? {
+    self.provider?.address
+  }
+
+  private var chainId: Int? {
+    self.provider?.chainId
+  }
+
+  public var client: ClientResponse? {
+    get async throws {
+      if self._client == nil {
+        self._client = try await self.getClient()
+      }
+      return self._client
+    }
+  }
+
+  /// Create an instance of a PortalApi class.
+  /// - Parameters:
+  ///   - apiKey: The Client API key. You can create one using Portal's REST API.
+  ///   - apiHost: (optional) The Portal API hostname.
+  ///   - provider: The PortalProvider instance to use for stateful Provider info (chainId, address)
+  public init(
+    apiKey: String,
+    apiHost: String = "api.portalhq.io",
+    provider: PortalProvider? = nil,
+    featureFlags: FeatureFlags? = nil,
+    requests: PortalRequests? = nil
+  ) {
+    self.apiKey = apiKey
+    self.baseUrl = apiHost.starts(with: "localhost") ? "http://\(apiHost)" : "https://\(apiHost)"
+    self.featureFlags = featureFlags
+    self.provider = provider
+    self.requests = requests ?? PortalRequests()
+    self.httpRequests = HttpRequester(baseUrl: self.baseUrl)
+  }
+
+  /*******************************************
```

Comment:
Everything in `Public functions` has been rewritten to use the new `PortalRequests` class instance to make HTTP requests, streamlining the process of handling the responses to only require data transformation.